include/util/ThreadPool.h

```diff
@@ -0,0 +1,148 @@
+/*******************************************************************************
+* Copyright 2009-2015 Juan Francisco Crespo Galán
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+******************************************************************************/
+
+#pragma once
+
+/**
+* @file ThreadPool.h
+* @ingroup util
+* The ThreadPool class.
+*/
+
+#include "Audaspace.h"
+
+#include <mutex>
+#include <condition_variable>
+#include <vector>
+#include <thread>
+#include <queue>
+#include <future>
+
+AUD_NAMESPACE_BEGIN
+/**
+* This represents pool of threads.
+*/
+class AUD_API ThreadPool
+{
+private:
+	/** 
+	* A queue of tasks.
+	*/
+	std::queue<std::function<void()>> m_queue;
+
+	/**
+	* A vector of thread objects.
+	*/
+	std::vector<std::thread> m_threads;
+
+	/**
+	* A mutex for synchronization.
+	*/
+	std::mutex m_mutex;
+
+	/**
+	* A condition variable used to stop the threads when there are no tasks.
+	*/
+	std::condition_variable m_condition;
+
+	/**
+	* Stop flag.
+	*/
+	bool m_stopFlag;
+
+	/**
+	* The number fo threads.
+	*/
+	unsigned int m_numThreads;
+
+	// delete copy constructor and operator=
+	ThreadPool(const ThreadPool&) = delete;
+	ThreadPool& operator=(const ThreadPool&) = delete;
+public:
+	/**
+	* Creates a new ThreadPool object.
+	* \param count The number of threads of the pool. It must not be 0.
+	*/
+	ThreadPool(unsigned int count) :
+		m_stopFlag(false), m_numThreads(count)
+	{
+		for(unsigned int i = 0; i < count; i++)
```

Comment:
Because it has a template method, but, thinking about it, I believe that including a .cpp file in the header could work. I'll try that in a bit.